src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/MobEventsUtils.java

```diff
@@ -0,0 +1,221 @@
+package com.minecolonies.coremod.entity.ai.mobs.util;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.ColonyView;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utils for Colony mob events
+ */
+public class MobEventsUtils
+{
+    private static final String BARBARIAN = "minecolonies.Barbarian";
+    private static final String ARCHER    = "minecolonies.ArcherBarbarian";
+    private static final String CHIEF     = "minecolonies.ChiefBarbarian";
+
+    private static final float  WHOLE_CIRCLE                 = 360.0F;
+    private static final float  HALF_A_CIRCLE                = 180F;
+    private static final int    MAX_SIZE                     = Configurations.maxBarbarianHordeSize;
+    private static final double BARBARIANS_MULTIPLIER        = 0.5;
+    private static final double ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    private static final double CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
+    private static final int    PREFERRED_MAX_HORDE_SIZE     = 40;
+    private static final int    PREFERRED_MAX_BARBARIANS     = 22;
+    private static final int    PREFERRED_MAX_ARCHERS        = 16;
+    private static final int    PREFERRED_MAX_CHIEFS         = 2;
+    private static       int    numberOfBarbarians           = 0;
+    private static       int    numberOfArchers              = 0;
+    private static       int    numberOfChiefs               = 0;
+    private static long TimeToRaid;
+
+    public static void barbarianEvent(final World world, final Colony colony)
+    {
+        numberOfSpawns(colony);
+
+        final BlockPos targetSpawnPoint = calculateSpawnLocation(world, colony);
+
+        if (Configurations.enableInDevelopmentFeatures)
+        {
+            LanguageHandler.sendPlayersMessage(
+              colony.getMessageEntityPlayers(),
+              "Horde Spawn Point: " + targetSpawnPoint);
+        }
+
+        LanguageHandler.sendPlayersMessage(
+          colony.getMessageEntityPlayers(),
+          "event.minecolonies.raidMessage");
+
+        BarbarianSpawnUtils.spawn(BARBARIAN, numberOfBarbarians, targetSpawnPoint, world);
+        BarbarianSpawnUtils.spawn(ARCHER, numberOfArchers, targetSpawnPoint, world);
+        BarbarianSpawnUtils.spawn(CHIEF, numberOfChiefs, targetSpawnPoint, world);
+    }
+
+    /**
+     * Sets the number of spawns for each barbarian type
+     *
+     * @param colony The colony to get the RaidLevel from
+     */
+    private static void numberOfSpawns(final Colony colony)
+    {
+        final int raidLevel = getColonyRaidLevel(colony);
+
+        if (colony.getCitizens().size() < 5)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)